addons/cluster-autoscaler/cluster-autoscaler.yaml

```diff
@@ -1,114 +1,227 @@
+# generated using below command:
+# helm template autoscaler/cluster-autoscaler --version 9.49.0 -n kube-system --name-template cluster-autoscaler \
+# --set cloudProvider=clusterapi,fullnameOverride=cluster-autoscaler,additionalLabels.k8s-app=cluster-autoscaler,additionalLabels.k8s-addon=cluster-autoscaler.addons.k8s.io,autoDiscovery.namespace=kube-system,extraEnv.CAPI_GROUP=cluster.k8s.io,"tolerations[0].effect=NoSchedule,tolerations[0].key=node-role.kubernetes.io/control-plane,affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key=node-role.kubernetes.io/control-plane,affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].operator=Exists" \
+# --set-string service.annotations."prometheus\.io/scrape"=true,service.annotations."prometheus\.io/port"=8085 | sed "s/x-k8s/k8s/g" | sed "/managed-by/d"
+
+# Must add/replace these in Deployment in the generated output for upgrades.
+        # - image: '{{ default (.InternalImages.Get "ClusterAutoscaler") .Params.CLUSTER_AUTOSCALER_IMAGE }}'
+        #   args:
+        #     - --skip-nodes-with-local-storage={{ default "true" .Params.CLUSTER_AUTOSCALER_SKIP_LOCAL_STORAGE }}
+        #     - --enforce-node-group-min-size={{ default "false" .Params.CLUSTER_AUTOSCALER_ENFORCE_NODE_GROUP_MIN_SIZE }}
+        #     - --balance-similar-node-groups={{ default "false" .Params.CLUSTER_AUTOSCALER_BALANCE_SIMILAR_NODE_GROUP }}
+        #     - --scale-down-utilization-threshold={{ default "0.5" .Params.CLUSTER_AUTOSCALER_SCALE_DOWN_UTIL_THRESHOLD }}
```

Comment:
Instead of providing instructions please write a Kustomization file that will automate them. For inspiration PTAL at https://github.com/kubermatic/kubeone/blob/main/addons/ccm-aws/Kustomization